03.bowling/bowling.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.split(',')
+shots = []
+
+scores.each do |s|
+  shots << if s == 'X'
+             10
+           else
+             s.to_i
+           end
+end
+
+frames = []
+i = 0
+while i < shots.length
+  if shots[i] == 10
+    frames << [10]
+    i += 1
+  else
+    frames << [shots[i], shots[i + 1]]
+    i += 2
+  end
```

Comment:
```suggestion
  length = shots[i] == 10 ? 1 : 2
  frames << shots[i, length]
  i += length
```

ちょっと上級者向けですがこんな風に書いても動きそうです